src/components/user-activity.ts

```diff
@@ -45,13 +45,15 @@ export interface UserActivity {
 export interface UserActivityTrackerConfig {
     inactiveAfterDays: number;
     minUserActiveDays: number;
+    debounceTimeMs:    number;
 }
 
 // eslint-disable-next-line @typescript-eslint/no-namespace,no-redeclare
 export namespace UserActivityTrackerConfig {
     export const DEFAULT: UserActivityTrackerConfig = {
         inactiveAfterDays: 31,
         minUserActiveDays: 3,
+        debounceTimeMs:    0,
```

Comment:
I have a fear this is still a severe footgun, and wonder if we should default it to something like 100ms to be on the safe side. Honestly I feel the logic here doesn't need to be close to real-time though, we could easily up this to a minute and not have a massively noticeable difference.